src/Frame.h

```diff
@@ -4,25 +4,31 @@
 #define frame_h
 
 #include <vector>
-using namespace std;
+#include <unordered_set>
+#include <memory>
 
 #include "Val.h"
 
+using namespace std;
```

Comment:
Please don't use "using namespace" in a header file, it'll propagate to everybody including it.